Base.rte/Activities/BunkerBreach.lua

```diff
@@ -2,213 +2,263 @@
-	local attackerLZ = SceneMan.Scene:GetArea("LZ Attacker");
-	local defenderLZ = SceneMan.Scene:GetOptionalArea("LZ Defender");	--Optional! To-do: define these in all Bunker Breach scenes?
-	self:SetLZArea(self.attackerTeam, attackerLZ);
-	if defenderLZ then
-		self:SetLZArea(self.defenderTeam, defenderLZ);
-	end
-
-	self.difficultyRatio = self.Difficulty/Activity.MAXDIFFICULTY;
-	self.CPUMaxDiggerCount = math.floor(self.difficultyRatio * 5);
-	-- Timers
-	self.checkTimer = Timer();
-	self.checkTimer:SetRealTimeLimitMS(1000);
-	self.CPUSpawnTimer = Timer();
-	self.CPUSpawnDelay = (40000 - self.difficultyRatio * 20000) * rte.SpawnIntervalScale;
-	self.CPUSearchRadius = math.sqrt(SceneMan.SceneWidth^2 + SceneMan.SceneHeight^2) * 0.5 * self.difficultyRatio;
-
-	--Set all actors in the scene to the defending team
-	for actor in MovableMan.AddedActors do
-		--To-do: allow attackers to spawn near the brain?
-		if actor.Team ~= self.defenderTeam then
-			MovableMan:ChangeActorTeam(actor, self.defenderTeam);
-		end
-	end
-
-	--Clear wonky default scene launch text
-	for player = Activity.PLAYER_1, Activity.MAXPLAYERCOUNT - 1 do
-		if self:PlayerActive(player) and self:PlayerHuman(player) then
-			FrameMan:ClearScreenText(self:ScreenOfPlayer(player));
-		end
-	end
+Add internal reinforcement doors by placing areas named:
+"Internal Reinforcements"
+You should make sure the area's boxes fit properly in a hallway - specifically you need to position the top left-corner of each box such that a 48 x 48 pixel background door can be safely placed with its top-left corner there. Note that positions will be snapped, to keep this from being too arduous.
+--]]
 
-	--CPU team setup
+function BunkerBreach:SetupAIVariables()
+	self.AI = {};
+	self.AI.isAttackerTeam = self.CPUTeam == self.attackerTeam;
+	self.AI.isDefenderTeam = self.CPUTeam == self.defenderTeam;
+	
 	if self.CPUTeam ~= Activity.NOTEAM then
-		self:SetTeamFunds(4000 + 8000 * math.floor(self.difficultyRatio * 5)/5, self.CPUTeam);
-		if (self.CPUTeam ~= self.defenderTeam) then
-			self.CPUSpawnDelay = self.CPUSpawnDelay * 0.5;
+		self.AI.difficultyRatio = self.Difficulty / Activity.MAXDIFFICULTY;
+		self.AI.maxDiggerCount = math.floor(self.AI.difficultyRatio * 5);
+		self.AI.spawnTimer = Timer();
+		self:CalculateAISpawnDelay(self.AI.isAttackerTeam); -- Make the first attacker spawn show up extra quickly, to minimize dead time at the start of the activity.
+
+		self:SetTeamFunds(4000 + ((8000 * math.floor(self.AI.difficultyRatio * (self.AI.isAttackerTeam and 8 or 4))) / 4), self.CPUTeam);
+
+		if self.AI.isAttackerTeam then
+			self.AI.maxCrabCount = 0;
+			self.AI.majorAttackTimer = Timer();
+			self.AI.initialMajorAttackDelay = self.Difficulty >= Activity.MEDIUMDIFFICULTY and (300000 - (self.AI.difficultyRatio * 200000)) or -1;  -- Only do major attacks if we're at or above medium difficulty.
```

Comment:
Never a fan of magic numbers :(

Reply:
I guess. It's all MS though, so it's not exactly magic? 30000 is 30 seconds, etc.
I can pull it out into some local variable if you think it's too egregious as-is.

Reply:
I had no idea that this was a time on first read to be honest 😄  I mainly gleamed over it, and clearly missed the "Delay" on the end of the variable. I'm not too fussed -

However, the code has 300,000, not 30,000.

Reply:
Ah yeah, I misread it here haha. 300 seconds is deliberate, cause it scales aggressively with ai difficulty ratio